src/distribution/lognormal.rs

```diff
@@ -0,0 +1,223 @@
+use distribution;
+use source::Source;
+
+/// A lognormal distribution.
+#[derive(Clone, Copy)]
+pub struct Lognormal {
+    mu: f64,
+    sigma: f64,
+}
+
+impl Lognormal {
+    /// Create a lognormal distribution with location `mu` and scale `sigma`.
+    ///
+    /// It should hold that `sigma > 0`.
+    #[inline]
+    pub fn new(mu: f64, sigma: f64) -> Self {
+        should!(sigma > 0.0);
+        Lognormal { mu: mu, sigma: sigma }
+    }
+
+    /// Return the location parameter.
+    #[inline(always)]
+    pub fn mu(&self) -> f64 { self.mu }
+
+    /// Return the scale parameter.
+    #[inline(always)]
+    pub fn sigma(&self) -> f64 { self.sigma }
+}
+
+impl Default for Lognormal {
+    #[inline]
+    fn default() -> Self {
+        Lognormal::new(0.0, 1.0)
+    }
+}
+
+impl distribution::Continuous for Lognormal {
+    #[inline]
+    fn density(&self, x: f64) -> f64 {
+        use std::f64::consts::PI;
+        should!(x > 0.0);
+        let &Lognormal { mu, sigma } = self;
+        (-(x.ln() - mu).powi(2) / (2.0 * sigma * sigma)).exp() / (x * sigma * (2.0 * PI).sqrt())
+    }
+}
+
+impl distribution::Distribution for Lognormal {
+    type Value = f64;
+
+    fn distribution(&self, x: f64) -> f64 {
+        distribution::Gaussian::new(self.mu, self.sigma).distribution(x.ln())
+    }
+}
+
+impl distribution::Entropy for Lognormal {
+    #[inline]
+    fn entropy(&self) -> f64 {
+        use std::f64::consts::PI;
+        (self.sigma * (self.mu + 0.5).exp() * (2.0 * PI).sqrt()).ln()
+    }
+}
+
+impl distribution::Inverse for Lognormal {
+    fn inverse(&self, p: f64) -> f64 {
+        should!(0.0 <= p && p <= 1.0);
+        distribution::Gaussian::new(self.mu, self.sigma).inverse(p).exp()
+    }
+}
+
+impl distribution::Kurtosis for Lognormal {
+    #[inline]
+    fn kurtosis(&self) -> f64 {
+        let s2 = self.sigma.powi(2);
+        (4.0 * s2).exp() + 2.0 * (3.0 * s2).exp() + 3.0 * (2.0 * s2).exp() - 6.0
+    }
+}
+
+impl distribution::Mean for Lognormal {
+    #[inline]
+    fn mean(&self) -> f64 {
+        (self.mu + self.sigma.powi(2) / 2.0).exp()
+    }
+}
+
+impl distribution::Median for Lognormal {
+    #[inline]
+    fn median(&self) -> f64 {
+        self.mu.exp()
+    }
+}
+
+impl distribution::Modes for Lognormal {
+    #[inline]
+    fn modes(&self) -> Vec<f64> {
+        let mode = (self.mu - self.sigma.powi(2)).exp();
+        vec![mode]
+    }
+}
+
+impl distribution::Sample for Lognormal {
+    #[inline]
+    fn sample<S>(&self, source: &mut S) -> f64 where S: Source {
+        use distribution::Inverse;
+        self.inverse(source.read::<f64>())
```

Comment:
Well, you can sample right from the gaussian that will be embedded into `Lognormal`.